File: src/logger.js
export class Logger {
  constructor(stdout) {
    this.stdout = stdout;
  }

  info(d) {
    this.stdout.write(
      JSON.stringify({
        info: d,
      }) + "\n"
    );
  }

  error(d) {
    this.stdout.write(
      JSON.stringify({
        error: d,
      }) + "\n"
    );
  }
}

File: logger.lua
local micro = import("micro")
local json = require"json"

local Logger = {}
Logger.__index = Logger

function Logger.new()
    return setmetatable({}, Logger)
end

function Logger:fromJson(output_string)
  local status, output = pcall(json.decode, output_string)

  if status then
    if output.info ~= nil or output.error ~= nil then
      micro.Log(output_string)

      return true
    end
  end

  return false
end

return Logger